pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -0,0 +1,252 @@
+
+        _label = f"┤↗{_plane}"
+
+        if decimals is not None:
+
+            def _format(x):
+                try:
+                    return format(qml.math.toarray(x), f".{decimals}f")
+                except ValueError:
+                    # If the parameter can't be displayed as a float
+                    return format(x)
+
+            _label += f"({float(_format(self.angle))})"
+
+        if self.postselect is not None:
+            _label += "₁" if self.postselect == 1 else "₀"
+
+        _label += "├" if not self.reset else "│  │0⟩"
+
+        return _label
+
+
+@_add_operation_to_drawer.register
+def _(op: ParametricMidMeasureMP, drawer, layer, _):
+    text = op.plane
+    drawer.measure(layer, op.wires[0], text=text)  # assume one wire
+
+    if op.reset:
+        drawer.erase_wire(layer, op.wires[0], 1)
+        drawer.box_gate(
+            layer + 1,
+            op.wires[0],
+            "|0⟩",
+            box_options={"zorder": 4},
+            text_options={"zorder": 5},
+        )
+
+
+def null_postprocessing(results):
+    """A postprocessing function returned by a transform that only converts the batch of results
+    into a result for a single ``QuantumTape``.
+    """
+    return results[0]
+
+
+@qml.transform
+def diagonalize_mcms(tape):
+    """Diagonalize any mid-circuit measurements in a parameterized basis into the computational basis.
+
+    Args:
+        tape (QNode or QuantumScript or Callable): The quantum circuit to modify the mid-circuit measurements of.
+
+    Returns:
+        qnode (QNode) or tuple[List[QuantumScript], function]: The transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    **Examples:**
+
+    This transform allows us to transform mid-circuit measurements into the measurement basis by adding
+    the relevant diagonalizing gates to the tape just before the measurement is performed.
+
+    .. code-block:: python3
+
+        from pennylane.ftqc import diagonalize_mcms, ParametricMidMeasureMP
+
+        dev = qml.device("default.qubit")
+
+        @diagonalize_mcms
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RY(x[0], wires=0)
+            ParametricMidMeasureMP(0, angle=x[1], plane="XY")
+            return qml.expval(qml.Z(0))
+
+    Applying the transform inserts the relevant gates before the measurement to allow
+    measurements to be in the Z basis, so the original circuit
+
+    >>> print(qml.draw(circuit, level=0)([np.pi/4, np.pi/4]))
+    0: ──RY(0.79)──┤↗ˣʸ(3.14)├─┤  <Z>
+
+    becomes
+
+    >>> print(qml.draw(circuit)([np.pi/4, np.pi/4]))
+    0: ──RY(0.79)──U(M0)──┤↗ᶻʸ(0.0)├─┤  <Z>
+    M0 =
+    [[ 0.70710678+0.j   0.5       -0.5j]
+     [ 0.70710678+0.j  -0.5       +0.5j]]
+
+    """
+
+    new_operations = []
+
+    for op in tape.operations:
+        # ToDo: maybe this in-place modification is a bad idea, but
+        #  its also weird to diagonalize it and not update it
```

Comment:
Hmmm, good point --- maybe fine for now, but in-place mods likely may come back to bite later.

Reply:
Updated this based on a suggestion from Mudit, so in the most recent implementation we don't have the in-place mutation of the MP anymore - see changes to the transform.

This comes at a slight cost, which is that any time we want to implement a new way of using a mid-circuit measurements in a function, we will have to update the transform. Right now we only pass them forward as input to the conditional of `cond`, so that's all the transform checks. But if we wanted to enable something like:

`qml.measure_xy(0, angle=np.pi/3 * (-1)**m)`

or anything else where we use a measurement output as input in another function, we have to add tracking that function to the transform.

Reply:
@lillian542 if we assume that the MCM will end up in `op.data`, we only need one extra logical branch :)

Reply:
Ah, good point - nice! 🚀

Reply:
But hypothetically, for example, if we were passing it into a correction tracking system that needed to accumulate measurement values and calculate fine corrections after execution entirely separate from existing PennyLane architecture, and we needed to update what results that tracked, we might also need to add it for that, depending when we are generating it. 

That's the only other use-case I can think of at the moment, but since we're trying to keep our options open in terms of what could need to happen, we should remember that for any process that needs this diagonalization, we have to consider where we are using these.